fluss-client/src/main/java/org/apache/fluss/client/utils/MetadataUtils.java

```diff
@@ -253,10 +256,14 @@ public NewTableMetadata(
         }
     }
 
-    public static ServerNode getOneAvailableTabletServerNode(Cluster cluster) {
+    public static @Nullable ServerNode getOneAvailableTabletServerNode(
+            Cluster cluster, Set<Integer> unavailableTabletServerIds) {
         List<ServerNode> aliveTabletServers = cluster.getAliveTabletServerList();
+        aliveTabletServers.removeIf(
+                serverNode -> unavailableTabletServerIds.contains(serverNode.id()));
```

Comment:
1. copy the `cluster.getAliveTabletServerList()` into a new list
2. call `removeIf` only when `unavailableTabletServerIds` is not empty.